magicctapipe/scripts/lst1_magic/coincident_events.py

```diff
@@ -0,0 +1,179 @@
+    """
+    
+    
+    coincidence_DL1_dir = target_dir+"/DL1/Observations"
+    if not os.path.exists(coincidence_DL1_dir+"/Coincident"):
+        os.mkdir(f"{coincidence_DL1_dir}/Coincident")
+    
+    for i in LST_runs:
+        lstObsDir = i[0].split("_")[0]+i[0].split("_")[1]+i[0].split("_")[2]
+        inputdir = f'/fefs/aswg/data/real/DL1/{lstObsDir}/v0.9/tailcut84'
+        outputdir = f'{coincidence_DL1_dir}/Coincident/{lstObsDir}'
+        list_of_subruns = np.sort(glob.glob(f"{inputdir}/dl1*Run*{i[1]}*.*.h5"))
+        if os.path.exists(f"{outputdir}/list_LST.txt"):
+            with open(f"{outputdir}/list_LST.txt", "a") as LSTdataPathFile:
+                for subrun in list_of_subruns:
+                    LSTdataPathFile.write(subrun+"\n") #If this files already exists, simply append the new information
+        else:
+            os.mkdir(outputdir)
+            f = open(f"{outputdir}/list_LST.txt", "w") #If the file list_LST.txt does not exist, it will be created here
+            for subrun in list_of_subruns:
+                f.write(subrun+"\n")
+            f.close()
+        
+     
+def bash_coincident(target_dir):
+
+    """
+    This function generates the bashscript for running the coincidence analysis.
+    
+    Parameters
+    ----------
+    target_dir: str
+        Path to the working directory
+    """
+
+    process_name = target_dir.split("/")[-2:][1]
+    
+    listOfNightsLST = np.sort(glob.glob(target_dir+"/DL1/Observations/Coincident/*"))
+    listOfNightsMAGIC = np.sort(glob.glob(target_dir+"/DL1/Observations/Merged/Merged*"))
+    
+    for nightMAGIC,nightLST in zip(listOfNightsMAGIC,listOfNightsLST):
+        process_size = len(np.genfromtxt(nightLST+"/list_LST.txt",dtype="str")) - 1
+        
+        f = open(f"LST_coincident_{nightLST.split('/')[-1]}.sh","w")
+        f.write("#!/bin/sh\n\n")
+        f.write("#SBATCH -p short\n")
+        f.write("#SBATCH -J "+process_name+"_coincidence\n")
+        f.write(f"#SBATCH --array=0-{process_size}%50\n")
+        f.write("#SBATCH -N 1\n\n")
+        f.write("ulimit -l unlimited\n")
+        f.write("ulimit -s unlimited\n")
+        f.write("ulimit -a\n\n") 
+
+        f.write(f"export INM={nightMAGIC}\n")
+        f.write(f"export OUTPUTDIR={nightLST}\n")
+        f.write("SAMPLE_LIST=($(<$OUTPUTDIR/list_LST.txt))\n")
+        f.write("SAMPLE=${SAMPLE_LIST[${SLURM_ARRAY_TASK_ID}]}\n")
+        f.write("export LOG=$OUTPUTDIR/coincidence_${SLURM_ARRAY_TASK_ID}.log\n")
+        f.write(f"conda run -n magic-lst python lst1_magic_event_coincidence.py --input-file-lst $SAMPLE --input-dir-magic $INM --output-dir $OUTPUTDIR --config-file {target_dir}/config_coincidence.yaml >$LOG 2>&1")
+        f.close()
+        
+
+
+def main():
+
+    """
+    Here we read the config_general.yaml file and call the functions defined above.
+    """
+    
+    
+    with open("config_general.yaml", "rb") as f:   # "rb" mode opens the file in binary format for reading
```

Comment:
Like it is now, this script must be called either from this directory only, otherwise it will not find the configuration file. I suggest to a) add an argument parser with which the user can pass the path to the configuration file and b) make this script a console script, so that it can be called as an executable from anywhere in the system

Reply:
I know how to pass to the scripts both configuration file (+ path) and MCP scripts paths; but what do you mean by point b)? (Sorry for ignorance)

Reply:
No worries, it is simple. You can add the scripts in setup.py, see the section console scripts. In that way, when calling pip install, pip will create executables calling the module/function specified. The executables will be put in the bin directory of the corresponding conda environment, so that they are added to the current path and make it possible to call them from anywhere in the system

Reply:
Thank you Alessio. I tried to implement it. Could you check it (in particular the console scripts section)?

Reply:
It seems so:

```bash
(magic-lst1) mpp@mpp-ThinkPad-L14-Gen-2 ~/Software/magic-cta-pipe $ (Torino) setting_up_config_and_dir -h
usage: setting_up_config_and_dir [-h] [--partial-analysis PARTIAL_ANALYSIS] [--config-file CONFIG_FILE]

optional arguments:
  -h, --help            show this help message and exit
  --partial-analysis PARTIAL_ANALYSIS, -p PARTIAL_ANALYSIS
                        You can type 'onlyMAGIC' or 'onlyMC' to run this script only on MAGIC or MC data, respectively.
  --config-file CONFIG_FILE, -c CONFIG_FILE
                        Path to a configuration file
(magic-lst1) mpp@mpp-ThinkPad-L14-Gen-2 ~/Software/magic-cta-pipe $ (Torino) merging_runs_and_splitting_training_samples -h
usage: merging_runs_and_splitting_training_samples [-h] [--config-file CONFIG_FILE]

optional arguments:
  -h, --help            show this help message and exit
  --config-file CONFIG_FILE, -c CONFIG_FILE
                        Path to a configuration file
(magic-lst1) mpp@mpp-ThinkPad-L14-Gen-2 ~/Software/magic-cta-pipe $ (Torino) coincident_events -h
usage: coincident_events [-h] [--config-file CONFIG_FILE]

optional arguments:
  -h, --help            show this help message and exit
  --config-file CONFIG_FILE, -c CONFIG_FILE
                        Path to a configuration file
(magic-lst1) mpp@mpp-ThinkPad-L14-Gen-2 ~/Software/magic-cta-pipe $ (Torino) stereo_events -h
usage: stereo_events [-h] [--config-file CONFIG_FILE]

optional arguments:
  -h, --help            show this help message and exit
  --config-file CONFIG_FILE, -c CONFIG_FILE
                        Path to a configuration file
```